include/lbann/data_readers/sample_list_jag_impl.hpp

```diff
@@ -329,6 +329,9 @@ inline void sample_list_jag::clear() {
   m_sample_list.clear();
 }
 
+template <class Archive> void sample_list_jag::serialize( Archive & ar ) {
```

Comment:
better make this inline

Reply:
Just FYI, function templates are automatically inline.

Reply:
I see. Good to know.

Reply:
Technically, I mean function templates are allowed to violate the ODR by default. _Specializations_ are not, so if you were to specialize this function, `inline` could (should) decorate the specializations. (Don't specialize this function -- overload it instead, but that's another issue.) If you want to actually force the compiler to inline the definition, from an ISO C++ level that's not possible, but compilers might have custom annotations to give programmers more influence.